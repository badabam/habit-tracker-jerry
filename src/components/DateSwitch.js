import React, { Component } from 'react'
import styled from 'react-emotion'
import moment from 'moment'

const Container = styled('section')`
  display: flex;
  font-family: sans-serif;
  align-items: center;
  justify-content: space-between;
  height: 40px;
  background: #fff;
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
`

export const SideButton = styled('button')`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  height: 100%;
  border: none;
  background: #eee;
  font-size: 1.3em;
`

export const DateText = styled('span')`
  font-weight: ${props => (props.isToday ? 'bold' : 'normal')};
`

export default class DateSwitch extends Component {
  state = {
    dayOffset: 0,
  }

  get currentDate() {
    return moment()
      .add(this.state.dayOffset, 'days')
      .format('DD.MM.YYYY')
  }

  moveDayLeft = () => {
    this.setState(
      state => ({
        dayOffset: state.dayOffset - 1,
      }),
      () => this.props.onUpdate(this.currentDate)
    )
  }

  moveDayRight = () => {
    this.setState(
      {
        dayOffset: this.state.dayOffset + 1,
      },
      () => this.props.onUpdate(this.currentDate)
    )
  }

  componentDidMount() {
    this.props.onUpdate(this.currentDate)
  }

  render() {
    const { dayOffset } = this.state

    return (
      <Container>
        <SideButton onClick={() => this.moveDayLeft()}>‹</SideButton>
        <DateText isToday={dayOffset === 0}>{this.currentDate}</DateText>
        <SideButton
          disabled={dayOffset === 0}
          onClick={() => this.moveDayRight()}
        >
          ›
        </SideButton>
      </Container>
    )
  }
}
