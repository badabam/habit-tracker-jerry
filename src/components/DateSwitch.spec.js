import { shallow, mount } from 'enzyme'
import moment from 'moment'
import React from 'react'
import toJson from 'enzyme-to-json'
import DateSwitch, { SideButton, DateText } from './DateSwitch'

describe('DateSwitch', () => {
  let wrapper
  let handleUpdate

  beforeEach(() => {
    jest.mock('moment', () => () => ({
      add: () => ({ format: () => '21.2.2018' }),
    }))
    handleUpdate = jest.fn()
    wrapper = mount(<DateSwitch onUpdate={handleUpdate} />)
  })
  afterEach(() => {
    handleUpdate.mockClear()
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('shows the date', () => {
    expect(wrapper.contains('21.2.2018')).toBe(true)
  })

  it('has a button to the right', () => {
    expect(wrapper.containsMatchingElement(<SideButton>‹</SideButton>)).toBe(
      true
    )
  })
  it('has a button to the right', () => {
    expect(wrapper.containsMatchingElement(<SideButton>›</SideButton>)).toBe(
      true
    )
  })
})
