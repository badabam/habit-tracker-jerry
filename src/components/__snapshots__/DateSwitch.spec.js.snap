// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSwitch renders correctly 1`] = `
.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 40px;
  background: #fff;
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 100%;
  border: none;
  background: #eee;
  font-size: 1.3em;
}

.emotion-1 {
  font-weight: bold;
}

<DateSwitch
  onUpdate={[Function]}
>
  <Styled(section)>
    <section
      className="emotion-3"
    >
      <Styled(button)
        onClick={[Function]}
      >
        <button
          className="emotion-0"
          onClick={[Function]}
        >
          ‹
        </button>
      </Styled(button)>
      <Styled(span)
        isToday={true}
      >
        <span
          className="emotion-1"
        >
          27.05.2018
        </span>
      </Styled(span)>
      <Styled(button)
        disabled={true}
        onClick={[Function]}
      >
        <button
          className="emotion-0"
          disabled={true}
          onClick={[Function]}
        >
          ›
        </button>
      </Styled(button)>
    </section>
  </Styled(section)>
</DateSwitch>
`;
